### ISSUE TITLE
same filename in two repositories shows once

### ISSUE THREAD

**Issue by sleep-walker** [orgzly-revived/orgzly-android-revived#432]
- I have searched for existing issues that may be the same as or related to mine. :heavy_check_mark: 

Please complete the following information:

    Device: Asus ROG Phone 6D
    Android Version: 14
    Orgzly Revived Version: 1.8.31 (fdroid)
    Sync repository type (directory/Dropbox/WebDAV/Git): 2 directories

I have 2 repositories - one personal, one work related, they are synchronized using Syncthing.

The problem I have is that there are two files with the same name in different locations. Orgzly Revived shows just one of them. The fact that they have different `#+TITLE` doesn't make a difference.

**Comment by amberin** [comment]
Yes, it's good to get a ticket for this very real bug. Notebooks are uniquely indexed by name in the database: https://github.com/orgzly-revived/orgzly-android-revived/blob/master/app/src/main/java/com/orgzly/android/db/entity/Book.kt#L11. I don't know the history behind this. Maybe it's for good reasons; maybe not.

**Comment by sleep-walker** [comment]
I don't know internals of this tool. But if you avoid various link types (or simply make the user responsible for not run into that), repository+filename or full path should be unique enough for DB.

**Comment by amberin** [comment]
> repository+filename or full path should be unique enough for DB.

The book's ID is the primary key. The issue would probably be that lots of current code looks up books by name only, leveraging that index. It is probably complicated by the fact that you can link to different repos or no repo, which should not change the book's identity.
